src/chunked.rs

```diff
@@ -99,6 +99,10 @@ impl<R: Read + Unpin> ChunkedDecoder<R> {
                 read += n;
                 let new_state = if new_current == len {
                     State::ChunkEnd
+                } else if n == 0 {
+                    // Unexpected end
+                    // TODO: do something?
+                    State::Done
```

Comment:
This seems like it should be an error. If no new data is passed we should indeed finish the stream, but because there's a mismatch between "expected length" and "actual length" it seems we should treat it as a protocol error.

Probably ignoring it for now though is fine; if the body ended prematurely any form of body parser will likely catch it (e.g. won't be valid json, csv, image, etc.).

So this seems good for now!